packages/shared/src/components/modals/RanksModal.tsx

```diff
@@ -197,6 +199,14 @@ export default function RanksModal({
   className,
   ...props
 }: RanksModalProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  useEffect(() => {
+    trackEvent({
+      event_name: 'show ranks modal',
+    });
+  }, []);
```

Comment:
We need to track it when it opens. `useEffect` might be tricking us.
You can check in the login modal how I implemented it.